lightning/src/ln/channelmanager.rs

```diff
@@ -2814,28 +2814,28 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		} else { false }
 	}
 
-	fn claim_funds_from_hop(&self, channel_state_lock: &mut MutexGuard<ChannelHolder<Signer>>, prev_hop: HTLCPreviousHopData, payment_preimage: PaymentPreimage) -> Result<(), Option<(PublicKey, MsgHandleErrInternal)>> {
+	fn claim_funds_from_hop(&self, channel_state_lock: &mut MutexGuard<ChannelHolder<Signer>>, prev_hop: HTLCPreviousHopData, payment_preimage: PaymentPreimage) -> (Option<u64>, Result<(), Option<(PublicKey, MsgHandleErrInternal)>>) {
```

Comment:
I think it would be nice to have a cleanup of the return values around the `claim_*` functions in this PR, similar to [this recent cleanup](https://github.com/rust-bitcoin/rust-lightning/pull/977/files#diff-8d28e96b6f27edb9a9739b2dce334f8906ec906155d421a7b5b02aa8e6c96057R304). They're getting pretty confusing :sweat_smile: